Genconfig panic with 'invalid memory address or nil pointer dereference'

### What happened

I tried running `talhelper genconfig` and it crashes with this error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x555a2e261909]
```

It was working for me yesterday. The issue appeared when first downgraded my talOS version then upgraded it. I did this because I wanted to try a custom patch of talOS to fix an unrelated issue with booting my rpi5 (config file homelab-pi-controlplane-2) from NVMe. And I wanted my currently running control plane node on amd64 architecture to match the talos version of the rpi5.

So, first I downgraded my nuc-lab-1 node with `talosctl upgrade -n 192.168.30.201 --image factory.talos.dev/metal-installer/613e1592b2da41ae5e265e8789429f22e121aab91cb4deb6bc3c0b6262961245:v1.12.7 --reboot-mode force`

Upgrade (downgrade) was successful. I tried the custom patched talos v1.12.7 from https://github.com/wheetazlab/talos-rpi-builder/releases/tag/v1.12.7-talos-k-rpi on the rpi5 (pi-controlplane-2 in my talconfig) but it didnt work.

Then I ran:
`talosctl upgrade -n 192.168.30.201 --image factory.talos.dev/metal-installer/613e1592b2da41ae5e265e8789429f22e121aab91cb4deb6bc3c0b6262961245:v1.13.2 --reboot-mode force`

Tried https://github.com/johnlaur/talos-builder/releases/tag/v1.13.2 on the pi. Boots up to maintenance mode so now it was time to generate my talconfig and apply it to the machine - and that's where this error appeared.

192.168.30.201 is the nuc-lab-1 node for reference.

I changed the `talosVersion` in talconfig to match the talos version I was installing. I don't know if this detail matters but perhaps it could help.

### Version
extra/talhelper v3.1.11, AUR
extra/talosctl 1.13.4-1.1, AUR
extra/kubectl 1.36.2-1, AUR

### Full debug log
```
2026/06/22 17:15:33 DEBUG start loading and validating config file
2026/06/22 17:15:33 DEBUG reading talconfig.yaml
2026/06/22 17:15:33 DEBUG substituting config file with environment variable
2026/06/22 17:15:33 DEBUG substituting relative paths with absolute paths
2026/06/22 17:15:33 DEBUG overriding global controlplane node config for nuc-lab-1
2026/06/22 17:15:33 DEBUG overriding global controlplane node config for pi-controlplane-2
2026/06/22 17:15:33 DEBUG overriding global worker node config for pi-worker-1
2026/06/22 17:15:33 DEBUG overriding global worker node config for pi-worker-2
2026/06/22 17:15:33 DEBUG overriding global worker node config for pi-worker-3
2026/06/22 17:15:33 DEBUG start validating talconfig file
2026/06/22 17:15:33 DEBUG validating config file for node nuc-lab-1
2026/06/22 17:15:33 DEBUG validating schematic with official Talos schematic for node nuc-lab-1
2026/06/22 17:15:33 DEBUG validating config file for node pi-controlplane-2
2026/06/22 17:15:33 DEBUG validating schematic with official Talos schematic for node pi-controlplane-2
2026/06/22 17:15:33 DEBUG validating config file for node pi-worker-1
2026/06/22 17:15:33 DEBUG validating schematic with official Talos schematic for node pi-worker-1
2026/06/22 17:15:33 DEBUG validating config file for node pi-worker-2
2026/06/22 17:15:33 DEBUG validating schematic with official Talos schematic for node pi-worker-2
2026/06/22 17:15:33 DEBUG validating config file for node pi-worker-3
2026/06/22 17:15:33 DEBUG validating schematic with official Talos schematic for node pi-worker-3
2026/06/22 17:15:33 DEBUG secret file is set to talsecret.sops.yaml
2026/06/22 17:15:33 DEBUG start generating config file
2026/06/22 17:15:33 DEBUG using secret file talsecret.sops.yaml
2026/06/22 17:15:33 DEBUG generating input file clusterName=homelab endpoint=https://192.168.30.201:6443 kubernetesVersion=1.34.1
2026/06/22 17:15:33 DEBUG setting input pod network to [10.244.0.0/16]
2026/06/22 17:15:33 DEBUG setting input service network to [10.96.0.0/12]
2026/06/22 17:15:33 DEBUG setting input additional machine cert SANs to []
2026/06/22 17:15:33 DEBUG setting input additional subject alt names to []
2026/06/22 17:15:33 DEBUG generating ./clusterconfig/homelab-nuc-lab-1.yaml for node nuc-lab-1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x555a2e261909]

goroutine 1 [running]:
github.com/siderolabs/talos/pkg/machinery/config/generate.(*Input).init(0x7cf586a2e00)
        /startdir/src/pkg/mod/github.com/siderolabs/talos/pkg/machinery@v1.14.0-alpha.1/config/generate/init.go:32 +0x229
github.com/siderolabs/talos/pkg/machinery/config/generate.(*Input).controlPlane(0x0?)
        /startdir/src/pkg/mod/github.com/siderolabs/talos/pkg/machinery@v1.14.0-alpha.1/config/generate/controlplane.go:14 +0x17
github.com/siderolabs/talos/pkg/machinery/config/generate.(*Input).Config(0x0?, 0x0?)
        /startdir/src/pkg/mod/github.com/siderolabs/talos/pkg/machinery@v1.14.0-alpha.1/config/generate/generate.go:124 +0x77
github.com/budimanjojo/talhelper/v3/pkg/talos.GenerateNodeConfig(0x7cf58640c00, 0x7cf586a2e00, 0x7cf5871f900, 0x0)
        /startdir/src/talhelper/pkg/talos/nodeconfig.go:32 +0x4f
github.com/budimanjojo/talhelper/v3/pkg/generate.GenerateConfig(0x7cf5869a008, 0x0, {0x555a2e325f8d, 0xf}, {0x555a2e32ece6?, 0x0?}, {0x555a2e313a04, 0x5}, 0x0, 0x0, ...)
        /startdir/src/talhelper/pkg/generate/config.go:43 +0x46d
github.com/budimanjojo/talhelper/v3/cmd.init.func8(0x7cf58634900?, {0x7cf58213280?, 0x4?, 0x555a2e312d3f?})
        /startdir/src/talhelper/cmd/genconfig.go:54 +0x33b
github.com/spf13/cobra.(*Command).execute(0x555a30385ee0, {0x7cf58213260, 0x1, 0x1})
        /startdir/src/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1019 +0xafb
github.com/spf13/cobra.(*Command).ExecuteC(0x555a30387020)
        /startdir/src/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
        /startdir/src/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
github.com/budimanjojo/talhelper/v3/cmd.Execute(...)
        /startdir/src/talhelper/cmd/root.go:61
main.main()
        /startdir/src/talhelper/main.go:10 +0x1b
```

## What steps I took to encounter this error
I extracted secrets from the already running 1-node (controlplane only) cluster with: `talosctl gen secrets --from-controlplane-config controlplane.yaml` which came out in the following format:

```yaml
cluster:
  id:
  secret:
secrets:
  bootstraptoken:
  secretboxencryptionsecret:
trustdinfo:
  token:
certs:
  etcd:
    crt:
    key:
  k8s:
    crt:
    key:
  k8saggregator:
    crt:
    key:
  k8sserviceaccount:
    key:
  os:
    crt:
    key:
```

Which I then had to reformat because `talosctl apply-config --file homelab-pi-controlplane-2 --config-patch talsecret.yaml -n <ip> --insecure` didnt recognize name of some keys from the talosctl generated secret file.

This is my current talsecret.yaml format:
```yaml
cluster:
  id: 
  secret: 
  token: 
  secretboxEncryptionSecret: 
  ca:
    crt: 
    key: 
  aggregatorCA:
    crt: 
    key: 
  serviceAccount:
    key: 
  etcd:
    ca:
      crt: 
      key: 
```
This is  my `talconfig.yaml` file:

Note that I want to wipe pi-controlplane-2 because I have tried installing on that machine multiple times so I need to wipe the disk on that one specifically

```yaml
---
clusterName: homelab
talosVersion: v1.13.2
kubernetesVersion: v1.34.1
endpoint: https://192.168.30.201:6443
domain: k8s.local
allowSchedulingOnControlPlanes: false
nodes:
  - hostname: nuc-lab-1
    ipAddress: 192.168.30.201
    controlPlane: true
    installDisk: /dev/nvme0n1
  - hostname: pi-controlplane-2
    ipAddress: 192.168.30.202
    controlPlane: true
    installDisk: /dev/nvme0n1
    patches:
      - |-
        machine:
          install:
            wipe: true
  - hostname: pi-worker-1
    ipAddress: 192.168.30.205
    controlPlane: false
    installDisk: /dev/nvme0n1
  - hostname: pi-worker-2
    ipAddress: 192.168.30.206
    controlPlane: false
    installDisk: /dev/nvme0n1
  - hostname: pi-worker-3
    ipAddress: 192.168.30.207
    controlPlane: false
    installDisk: /dev/nvme0n1
controlPlane:
  nodeLabels:
    node.kubernetes.io/exclude-from-external-load-balancers: ""
  nameservers:
    - 1.1.1.1
    - 8.8.8.8

worker:
  nameservers:
    - 1.1.1.1
    - 8.8.8.8

patches:
  - |-
    machine:
      kubelet:
        defaultRuntimeSeccompProfileEnabled: true # Enable container runtime default Seccomp profile.
        disableManifestsDirectory: true # The `disableManifestsDirectory` field configures the kubelet to get static pod manifests from the /etc/kubernetes/manifests directory.
        extraMounts:
          - destination: /var/lib/longhorn
            type: bind
            source: /var/lib/longhorn
            options:
              - bind
              - rshared
              - rw
      features:
        rbac: true # Enable role-based access control (RBAC).
        apidCheckExtKeyUsage: true # Enable checks for extended key usage of client certificates in apid.
      sysctls:
        # Longhorn requires this
        vm.nr_hugepages: "1024"
      kernel:
        modules:
          # longhorn V2 data engine requirements
          - name: nvme_tcp
          - name: vfio_pci
    cluster:
      # Cilium requires this
      network:
        cni:
            name: none
      # disable kube-proxy  - required for Cilium with support for GatewayAPI
      proxy:
        disabled: true
```

--- a/pkg/talos/input.go
+++ b/pkg/talos/input.go
@@ -1,6 +1,7 @@
 package talos
 
 import (
+	"bytes"
 	"fmt"
 	"log/slog"
 
@@ -41,7 +42,10 @@ func NewClusterInput(c *config.TalhelperConfig, secretFile string, mode string) 
 			return nil, err
 		}
 
-		err = yaml.Unmarshal(decrypted, &sb)
+		dec := yaml.NewDecoder(bytes.NewReader(decrypted))
+		dec.KnownFields(true)
+
+		err = dec.Decode(&sb)
 		if err != nil {
 			return nil, err
 		}

